pkg/dyninst/irgen/irgen.go

```diff
@@ -1794,24 +1795,76 @@ func populateEventExpressions(
 		default:
 			panic(fmt.Sprintf("unexpected event kind: %v", event.Kind))
 		}
-		variableSize := variable.Type.GetByteSize()
-		expr := &ir.RootExpression{
-			Name:   variable.Name,
-			Offset: uint32(0),
-			Kind:   variableKind,
-			Expression: ir.Expression{
-				Type: variable.Type,
-				Operations: []ir.ExpressionOp{
-					&ir.LocationOp{
-						Variable: variable,
-						Offset:   0,
-						ByteSize: uint32(variableSize),
-					},
-				},
-			},
-		}
+		expr := newRootExpressionForVariable(variable, variableKind)
 		expressions = append(expressions, expr)
 	}
+
+	// Process template segments if template exists
+	var variableExpressionSet map[string]int
```

Comment:
what is this?

Reply:
It maps expression names to their index within the event. It enables us to not create duplicate expressions when processing template segments.